build-tools/packages/build-cli/src/commands/release/prepare.ts

```diff
@@ -0,0 +1,266 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { MonoRepo, Package } from "@fluidframework/build-tools";
+import async from "async";
+import chalk from "chalk";
+import execa from "execa";
+import { ResetMode } from "simple-git";
+
+import { findPackageOrReleaseGroup, packageOrReleaseGroupArg } from "../../args.js";
+import {
+	BaseCommand,
+	type Context,
+	Repository,
+	getPreReleaseDependencies,
+} from "../../library/index.js";
+
+type CheckFunction = (
+	context: Context,
+	rgOrPkg: MonoRepo | Package,
+	print?: boolean,
+) => Promise<CheckResult>;
+
+interface CheckResult {
+	result: boolean;
+	message?: string;
+	fixCommand?: string;
+	fatal?: boolean;
+}
+
+async function checkNoLocalChanges(context: Context): Promise<CheckResult> {
```

Comment:
Since you are using the function syntax not the lambda syntax here, typescriopt doesn't provide a clear way to indicate this function implements the CheckFunction type (at least I think it might be intending that). A doc comment with a link would be a nice way to communicate that.